plugins/callback/logstash.py

```diff
@@ -1,57 +1,86 @@
-# (C) 2016, Ievgen Khmelenko <ujenmr@gmail.com>
+# (C) 2020, Yevhen Khmelenko <ujenmr@gmail.com>
 # (C) 2017 Ansible Project
 # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
-DOCUMENTATION = '''
-    author: Unknown (!UNKNOWN)
+DOCUMENTATION = r'''
     callback: logstash
     type: notification
+    author: Yevhen Khmelenko <ujenmr@gmail.com>
     short_description: Sends events to Logstash
     description:
       - This callback will report facts and task events to Logstash https://www.elastic.co/products/logstash
+    version_added: "2.3"
     requirements:
       - whitelisting in configuration
       - logstash (python library)
     options:
       server:
         description: Address of the Logstash server
-        env:
-          - name: LOGSTASH_SERVER
         ini:
           - section: callback_logstash
             key: server
-            version_added: 1.0.0
+        env:
+          - name: LOGSTASH_SERVER
         default: localhost
       port:
         description: Port on which logstash is listening
-        env:
-            - name: LOGSTASH_PORT
         ini:
           - section: callback_logstash
             key: port
-            version_added: 1.0.0
+        env:
+          - name: LOGSTASH_PORT
         default: 5000
+      pre_command:
+        description: Executes command before run and result put to ansible_pre_command_output field
+        ini:
+          - section: callback_logstash
+            key: pre_command
+        env:
+          - name: LOGSTASH_PRE_COMMAND
+        version_added: "2.6"
       type:
         description: Message type
-        env:
-          - name: LOGSTASH_TYPE
         ini:
           - section: callback_logstash
             key: type
-            version_added: 1.0.0
```

Comment:
This also needs to stay.

Reply:
added with https://github.com/ansible-collections/community.general/pull/641/commits/f32be815f0cb7f37bec0e72d752339d7fa709792